bindings/app.cpp

```diff
@@ -21,6 +21,20 @@ App::App(QString hash, QObject *parent) : QObject(parent), m_hash(hash)
     }
 }
 
+
+App::App(Dna *dna, QObject *parent) : QObject(parent)
+{
+    m_dna = dna;
+    if(m_dna){
+        // need to clone a dna object because holochain_new() consumes it..
+        // Might wanna change that on the rust side?
```

Comment:
Because this happens only once, not an efficiency problem, so I wouldn't worry too much about changing it.  I forget if the guideline is that you should take a reference if possible and clone it at the receiving end if you need to, or the other way around...

Reply:
Well, this is unexpected behavior in a C context and it did cost me several hours of weird pointer debugging to find out that this is the cause... The user story would start with: "As a native devloper who uses the C bindings...".